torchtitan/parallelisms/parallelize_llama.py

```diff
@@ -72,9 +72,13 @@ def parallelize_llama(
         apply_compile(model)
 
     if parallel_dims.dp_enabled:
-        if parallel_dims.dp_type == "fsdp":
-            dp_mesh = world_mesh["dp"] if world_mesh.ndim > 1 else world_mesh
-            assert dp_mesh.mesh_dim_names == ("dp",), dp_mesh.mesh_dim_names
+        if parallel_dims.dp_type in ("fsdp", "hsdp"):
+            if parallel_dims.dp_type == "hsdp":
+                dp_mesh = world_mesh["dp_replicate", "dp_shard"]
```

Comment:
to educate me: so FSDP would automatically work for HSDP if a 2d mesh is given -- to replicate in the outer mesh and shard in the inner mesh?

Reply:
Yes

Reply:
> to educate me: so FSDP would automatically work for HSDP if a 2d mesh is given -- to replicate in the outer mesh and shard in the inner mesh?

Just leave a reference here: https://github.com/pytorch/pytorch/blob/16de25b1dc3af7cc3efdb24e5637a62b8c329881/torch/distributed/_composable/fsdp/fully_shard.py#L112-L115
and
https://github.com/pytorch/pytorch/blob/16de25b1dc3af7cc3efdb24e5637a62b8c329881/torch/distributed/_composable/fsdp/_fsdp_param.py#L320-L322